From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TominoLP <erfinder.tom@gmail.com>
Date: Sun, 23 Oct 2022 15:57:50 +0200
Subject: [PATCH] Biom api fix


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java b/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
index 62bca85da6c5d9877e21fecb702370506ddf671c..4c97c30cc440fd98387138eaf66856a334dccbaa 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
@@ -216,8 +216,14 @@ public abstract class CraftRegionAccessor implements RegionAccessor {
     public Biome getComputedBiome(int x, int y, int z) {
         return CraftBlock.biomeBaseToBiome(this.getHandle().registryAccess().registryOrThrow(Registry.BIOME_REGISTRY), this.getHandle().getBiome(new BlockPos(x, y, z)));
     }
-    // Paper end
+    // Spectral start
+    @Override
+    public String getRegisteryBiom(int x, int y, int z) {
+        return this.getHandle().registryAccess().registryOrThrow(Registry.BIOME_REGISTRY).getKey(this.getHandle().getNoiseBiome(x >> 2, y >> 2, z >> 2).value()).toString();
+    }
+    // Spectral end
 
+    // Paper end
     @Override
     public void setBiome(Location location, Biome biome) {
         this.setBiome(location.getBlockX(), location.getBlockY(), location.getBlockZ(), biome);
diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
index 0d47460494135d4ec4c95260de033e054c2f0404..304cd6ae60bc70f48aba7716595642603f331eb8 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
@@ -5,7 +5,6 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
-import java.util.Objects;
 import java.util.stream.Collectors;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
@@ -348,6 +347,15 @@ public class CraftBlock implements Block {
     }
     // Paper end
 
+    // Spectral start
+    @Override
+    public String getRegistryBiome() {
+        net.minecraft.world.level.biome.Biome biome = this.getCraftWorld().getHandle().getChunkAt(this.position).getNoiseBiome(this.position.getX() >> 2, this.position.getY() >> 2, this.position.getZ() >> 2).value();
+        net.minecraft.core.Registry<net.minecraft.world.level.biome.Biome> reg = ((org.bukkit.craftbukkit.CraftServer) Bukkit.getServer()).getServer().registryHolder.registryOrThrow(net.minecraft.core.Registry.BIOME_REGISTRY);
+        return reg.getKey(biome).toString();
+    }
+    // Spectral end
+
     @Override
     public void setBiome(Biome bio) {
         this.getWorld().setBiome(this.getX(), this.getY(), this.getZ(), bio);
